### DISCUSSION TITLE
Is it possible to set Python setup-related environment variables?

### DISCUSSION THREAD

**Discussion by rtpg** [pyodide/pyodide#3869] [Q&A]
For reasons I would like to set `PYTHONHASHSEED`, but am not seeing an obvious way of doing this in Pyodide. Is this an option in the API?

Alternateively, is there an existing place in the build process where we are setting up environment variables to be used?

**Comment by hoodmane** [comment]
Surprisingly `loadPyodide` doesn't have an argument that you can use to pass environment variables. It certainly should have this. It does have the `args` argument that lets you pass extra command line flags to the interpreter... it seems that with `args: ["-R"]` you can override a value in `PYTHONHASHSEED` and force randomization, but there is no command line option to turn off hash randomization.

> is there an existing place in the build process where we are setting up environment variables to be used?

Give me a minute and I will make a PR to add environment variables to the `loadPyodide` arguments. If you are willing to make a custom build, you can backport my patch or add your own similar modifications.

**Reply by hoodmane** [reply]
I can confirm that on #3870, it is possible to get consistent hashing between runs with `PYTHONHASHSEED`. The hashes do not agree with native linux python3.11 though.
